pkgs/native_assets_builder/lib/src/dependencies_hash_file/dependencies_hash_file.dart

```diff
@@ -3,19 +3,22 @@
 // BSD-style license that can be found in the LICENSE file.
 
 import 'dart:convert';
-import 'dart:io';
+import 'dart:io' show Platform;
 import 'dart:typed_data';
 
 import 'package:crypto/crypto.dart';
+import 'package:file/file.dart';
 
 import '../utils/file.dart';
 import '../utils/uri.dart';
 
 class DependenciesHashFile {
-  DependenciesHashFile({
+  DependenciesHashFile(
+    this._fileSystem, {
```

Comment:
`file.fileSystem` already gives access to the file system. So this param is redundant. Either

1. remove the param and the field
2. remove the param, and init the field with `file.fileSystem` or,
3. add an assert that the file system is equal.

Reply:
I choose alternative option 4: Encapsulate all operations on the hashes file inside the `DependenciesHashFile` and make outside world only give it `FileSystem` and `Uri` and ask the `DependenciesHashFile` to check for existence, deleting. That seems much cleaner.